pokete_classes/learnattack.py

```diff
@@ -3,10 +3,36 @@
 import time
 import random
 import scrap_engine as se
-from pokete_general_use_fns import std_loop
+from pokete_general_use_fns import std_loop, liner
 from .input import ask_bool, ask_ok
-from .ui_elements import ChooseBox
+from .ui_elements import ChooseBox, Box
 from .detail import Detail
+from .color import Color
+from .attack import Attack
+
+
+class AttackInfo(Box):
+    """Gives information about a certain attack"""
+
+    def __init__(self, attack, p_d, _map):
+        obj = Attack(attack, p_d)
+        desc_label = se.Text(liner(obj.desc, 40))
+        super().__init__(4 + len(desc_label.text.split("\n")),
+                         sorted(len(i) for i in
+                            desc_label.text.split("\n")
+                            + [obj.label_type.text,
+                               obj.label_factor.text])[-1] + 4, obj.name,
+                         "q:close")
+        self.map = _map
+        self.add_ob(obj.label_type, 2, 1)
+        self.add_ob(obj.label_factor, 2, 2)
+        self.add_ob(desc_label, 2, 3)
```

Comment:
Rename `obj` to `atc`, as this gets information about an attack and not about an "unknown" object:

```suggestion
        atc = Attack(attack)
        desc_label = se.Text(liner(atc.desc, 40))
        super().__init__(4 + len(desc_label.text.split("\n")),
                         sorted(len(i) for i in
                            desc_label.text.split("\n")
                            + [atc.label_type.text,
                               atc.label_factor.text])[-1] + 4, atc.name,
                         "q:close")
        self.map = _map
        self.add_ob(atc.label_type, 2, 1)
        self.add_ob(atc.label_factor, 2, 2)
        self.add_ob(desc_label, 2, 3)
```